gbrl/src/cpp/utils.cpp

```
@@ -111,4 +124,52 @@ int count_distinct(T *arr, int n) {
+ * 
+ * Specialized version of selective_copy for character arrays with fixed-size
+ * strings (MAX_CHAR_SIZE bytes each). Uses memcpy for efficient copying.
+ * Parallelized with OpenMP.
+ * 
+ * @param num_indices Number of strings to copy
+ * @param indices Array of source indices to copy from
+ * @param dest Destination character array
+ * @param src Source character array
+ * @param elements_dim Number of strings per indexed element
+ */
+void selective_copy_char(const int num_indices, const int* indices, char* dest, const char* src, const int elements_dim){
```

Cppcheck (2.19.0)

[style] The function 'selective_copy_char' is never used.

(unusedFunction)